src/providers/ogr/qgsgeopackagedataitems.cpp

```diff
@@ -370,22 +369,39 @@ QList<QAction *> QgsGeoPackageVectorLayerItem::actions()
 
 void QgsGeoPackageVectorLayerItem::deleteLayer()
 {
-  if ( QMessageBox::question( nullptr, QObject::tr( "Delete Layer" ),
-                              QObject::tr( "Are you sure you want to delete layer '%1' from GeoPackage?" ).arg( mName ),
-                              QMessageBox::Yes | QMessageBox::No, QMessageBox::No ) != QMessageBox::Yes )
-    return;
-
-  QString errCause;
-  bool res = ::deleteLayer( mUri, errCause );
-  if ( !res )
+  // Check if the layer is in the registry
+  const QgsMapLayer *projectLayer = nullptr;
+  Q_FOREACH ( const QgsMapLayer *layer, QgsProject::instance()->mapLayers() )
```

Comment:
Someone decided to deprecated it but no clear migration path was outlined. But it was far from being an unanimous decision, mostly because the proposed replacement (range based for loops) don't handle implicit sharing. At the same time, `Q_FOREACH` doesn't have any major problems.
Everybody is welcome to use range based for-loops instead of `Q_FOREACH` or even propose a solution with which we can move towards a `Q_FOREACH`-free world.